promql/parse.go

```diff
@@ -1125,6 +1161,12 @@ func (p *parser) checkType(node Node) (typ ValueType) {
 			p.errorf("unary expression only allowed on expressions of type scalar or instant vector, got %q", documentedType(t))
 		}
 
+	case *SubqueryExpr:
+		ty := p.checkType(n.Expr)
+		if ty != ValueTypeScalar && ty != ValueTypeVector {
+			p.errorf("expected scalar or vector selector for subquery %q, got %s instead", n.String(), ty)
```

Comment:
Scalars don't make sense for subqueries, it has to be an instant vector

Reply:
Why? We could do a subquery with scalar()